### PR TITLE
Add TiDB to the list of vendor supported adapters

### PR DESCRIPTION
**Pull request by Daemonxiao**
## Description & motivation
Add TiDB adapter. 
[TiDB](https://github.com/pingcap/tidb) (/’taɪdiːbi:/, "Ti" stands for Titanium) is an open-source, distributed, NewSQL database that supports Hybrid Transactional and Analytical Processing (HTAP) workloads.

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [x] No: please ensure the base branch is `current`

## Checklist
If you added new pages (delete if not applicable):
- [x] The page has been added to `website/sidebars.js`
- [x] The new page has a unique filename

### CODE REVIEW

File: website/docs/reference/warehouse-profiles/tidb-profile.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+title: "TiDB Profile"
+---
+
+:::info Vendor-supported plugin
+
+Some core functionality may be limited. If you're interested in contributing, check out the source code for each repository listed below.
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/tidb-profile.md]: perhaps it's worth linking back to your readme about what is and isn't supported?

File: website/docs/reference/warehouse-profiles/tidb-profile.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    dev:
+      type: tidb
+      server: 127.0.0.1
+      port: 4000
+      schema: database_name
+      username: tidb_username
+      password: tidb_password
+```
+
+</File>
```
Comment (dataders) [inline website/docs/reference/warehouse-profiles/tidb-profile.md:48]: there's a lot of great information in the repo readme that would benefit from being here!

**Review by dataders — CHANGES_REQUESTED** [review]
@Daemonxiao this is awesome -- thanks!

Two suggestions:
1. can you please make this page the single-source-of-truth as opposed to your adapter repo's `README.md`? there's more info there that I think would benefit from being here.
2. is there any functionality about using TiDB that users might want to know about? for example, see [teradata's](https://docs.getdbt.com/reference/resource-configs/teradata-configs) and [firebolt's](https://docs.getdbt.com/reference/resource-configs/firebolt-configs) config pages

File: website/docs/reference/warehouse-profiles/tidb-profile.md (hunk 1 above)
Comment (Daemonxiao) [reply website/docs/reference/warehouse-profiles/tidb-profile.md]: Thanks to @dataders for review. I have linked the relevant documentation.

File: website/docs/reference/warehouse-profiles/tidb-profile.md (hunk 2 above)
Comment (Daemonxiao) [reply website/docs/reference/warehouse-profiles/tidb-profile.md:48]: Thanks. We are quite willing to make this page the source of truth.
We have added some key content to the page to help users get more useful information on the page.

Now, dbt-tidb mainly implements dbt configuration without additional functionsality. If we add any new func later, we will add a copy of config pagel.

File: website/docs/reference/warehouse-profiles/tidb-profile.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
+---
+title: "TiDB Profile"
+---
+
+:::info Vendor-supported plugin
+
+Some [core functionality](https://github.com/pingcap/dbt-tidb/blob/main/README.md#supported-features) may be limited. 
+If you're interested in contributing, check out the source code for each repository listed below.
```
Comment (zhangyangyu) [inline website/docs/reference/warehouse-profiles/tidb-profile.md]:
```suggestion
If you're interested in contributing, check out the source code repository listed below.
```

**Review by dataders — APPROVED** [review]
perfect! I'm happy with this as a first version. Please share new dbt-tidb's user experience with us. We can make a channel in dbt Slack too if you'd like. Please reach out to me on dbt Slack if you have anymore questions.

**Pull request merged**